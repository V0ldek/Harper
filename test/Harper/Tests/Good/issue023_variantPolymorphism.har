value T = {
    variant A = {
        foo :: a -> a;
        foo x = x;
    };

    variant B = {
        foo :: a -> Integer;
        foo x = 42;
    };
};

// The type of foo is unified to be Integer -> Integer.

main :: sideeffect -> ();
main = {
  (a :: T) = val A {};
  (b :: T) = val B {};

  eval printLn (a.foo 17) ();
  eval printLn (b.foo 17) ();
};