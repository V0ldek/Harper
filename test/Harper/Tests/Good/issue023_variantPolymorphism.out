17
42

Execution ended with value: ()